nix-tools/static/zipped.nix

```diff
@@ -0,0 +1,97 @@
+inputs: pkgs:
+
+let
+
+  makeZippedTools = { customPkgs, clearStripDebugFlags ? false }: 
+    let 
+      hsPkgs = (import ./project.nix inputs customPkgs).hsPkgs;
+      
+      maybeClearStripDebugFlags = drv: 
+        if clearStripDebugFlags then 
+          drv.overrideDerivation (_: { stripDebugFlags = []; })
+        else 
+          drv;
+
+      getNixToolsExe = name: hsPkgs.nix-tools.components.exes.${name};
+      getStrippedNixToolsExe = name: maybeClearStripDebugFlags (getNixToolsExe name);
+
+      strippedNixToolsComponents = map getStrippedNixToolsExe [
+        "cabal-name" 
+        "cabal-to-nix" 
+        "hackage-to-nix" 
+        "hashes-to-nix" 
+        "lts-to-nix" 
+        "make-install-plan" 
+        "plan-to-nix" 
+        "stack-repos" 
+        "stack-to-nix" 
+        "truncate-index" 
+      ];
+    in 
+      customPkgs.packaging.asZip {
+        name = "${customPkgs.hostPlatform.system}-nix-tools-static";
+        drvs' = [ 
+          hsPkgs.cabal-install.components.exes.cabal 
+          hsPkgs.hpack.components.exes.hpack 
+        ] ++ strippedNixToolsComponents;
+      };
+
+  
+  zippedToolsNoIfdFor = fragment-name: 
```

Comment:
We should document this in detail, why we need this, and what the issue with keeping IFDs is, and why using recursive-nix helps us here.